Source/JavaScriptCore/assembler/MacroAssemblerX86Common.cpp

```diff
@@ -810,18 +810,23 @@ void MacroAssemblerX86Common::collectCPUFeatures()
             s_sse4_2CheckState = (cpuid[2] & (1 << 20)) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
             s_popcntCheckState = (cpuid[2] & (1 << 23)) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
             s_avxCheckState = (cpuid[2] & (1 << 28)) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+            s_avxCheckState = (cpuid[2] & (1 << 28)) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
         }
 #if OS(DARWIN)
         {
             uint32_t val = 0;
             size_t valSize = sizeof(val);
             int rc = sysctlbyname("hw.optional.bmi1", &val, &valSize, nullptr, 0);
             s_bmi1CheckState = (rc >= 0 && val) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
+
+            rc = sysctlbyname("hw.optional.avx2_0", &val, &valSize, nullptr, 0);
+            s_avx2CheckState = (rc >= 0 && val) ? CPUIDCheckState::Set : CPUIDCheckState::Clear;
```

Comment:
Looks good. We are not using `getCPUID(0x7)` because it slows down hypervisors (VMExit happens).